### PR TITLE
I18n cloudwatch client

### PR DESCRIPTION
**Pull request by annaxuphoto**
## PR Checklist:

- [ ] Tests provide adequate coverage
- [ ] Privacy and Security impacts have been assessed
- [ ] Code is well-commented
- [ ] New features are translatable or updates will not break translations
- [ ] Relevant documentation has been added or updated
- [ ] User impact is well-understood and desirable
- [ ] Pull Request is labeled appropriately
- [ ] Follow-up work items (including potential tech debt) are tracked and linked

### CODE REVIEW

File: bin/i18n/i18n_metrics.rb [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+      attr_accessor :machine_id
+    end
+
+    def report_runtime(addtl_dimensions, value)
+      log_metric(:Runtime, addtl_dimensions, value)
+    end
+
+    def get_machine_id
+      begin
+        metadata_endpoint = 'http://169.254.169.254/latest/meta-data/'
```
Comment (artem-vavilov) [inline bin/i18n/i18n_metrics.rb:18]: It would be nice to move this URL to `config.yml.erb`

File: bin/i18n/i18n_metrics.rb [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+        return Net::HTTP.get(URI.parse(metadata_endpoint + 'instance-id'))
+      rescue
+        return "local_machine"
```
Comment (artem-vavilov) [inline bin/i18n/i18n_metrics.rb:21]: There is no need to put `return` at the end of the method
```suggestion
        Net::HTTP.get(URI.parse(metadata_endpoint + 'instance-id'))
      rescue
        'local_machine'
```

File: bin/i18n/i18n_metrics.rb [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+require 'net/http'
+require 'socket'
+
+module I18n
+  module Metrics
+    class << self
+      attr_accessor :machine_id
+    end
+
+    def report_runtime(addtl_dimensions, value)
```
Comment (artem-vavilov) [inline bin/i18n/i18n_metrics.rb:12]:
```suggestion
    def self.report_runtime(addtl_dimensions, value)
```

File: bin/i18n/i18n_metrics.rb [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+      begin
+        metadata_endpoint = 'http://169.254.169.254/latest/meta-data/'
+        return Net::HTTP.get(URI.parse(metadata_endpoint + 'instance-id'))
+      rescue
+        return "local_machine"
+      end
+    end
+
+    def log_metric(metric_name, addtl_dimensions, value)
+      machine_id = I18n::Metrics.machine_id ||= get_machine_id
```
Comment (artem-vavilov) [inline bin/i18n/i18n_metrics.rb:26]: 1. There is no need to add `return` at the end of a method
2. https://github.com/code-dot-org/code-dot-org/blob/staging/STYLEGUIDE.md#ruby-string-quotes
```suggestion
    def self.machine_id
      @machine_id ||= begin
        metadata_endpoint = 'http://169.254.169.254/latest/meta-data/'
        Net::HTTP.get(URI.parse(metadata_endpoint + 'instance-id'))
      rescue
        'local_machine'
      end
    end

    def self.log_metric(metric_name, addtl_dimensions, value)
```

File: bin/i18n/i18n_metrics.rb (hunk 1 above)
Comment (artem-vavilov) [inline bin/i18n/i18n_metrics.rb:18]: It would be nice to move this to `config.yml.erb`

File: bin/i18n/i18n_metrics.rb (hunk 3 above)
Comment (artem-vavilov) [inline bin/i18n/i18n_metrics.rb:12]:
```suggestion
    def self.report_runtime(addtl_dimensions, value)
```

File: bin/i18n/i18n_metrics.rb (hunk 4 above)
Comment (annaxuphoto) [reply bin/i18n/i18n_metrics.rb:26]: yep fair! I find it more readable with a return statement but happy to follow Ruby convention as well

File: bin/i18n/i18n_metrics.rb (hunk 3 above)
Comment (annaxuphoto) [reply bin/i18n/i18n_metrics.rb:12]: Does self mean anything if these methods live in a module, not a class?

Comment (artem-vavilov) [reply bin/i18n/i18n_metrics.rb:12]: It becomes defined on the module’s [eigenclass](https://medium.com/@ethan.reid.roberts/rubys-anonymous-eigenclass-putting-the-ei-in-team-ebc1e8f8d668).

If you want to call it from the module as `18n::Metrics.report_runtime`, you have 3 options for defining the method: https://www.mendelowski.com/docs/ruby/self-vs-extend-self-vs-module-function/

I would recommend using the `self` definition for consistency on the project.

**Pull request closed without merging**